azavea/fb-gender-survey-dashboard
# Process 2021 config data

## Overview

Updates the data processing script to handle the new 2021 column names and use the 2021 config data.

Updates the categories to: Norms, Access, and Agency (A); Time Spent and Care (B ); COVID's impact (C); and Demographics (D).

Note that the full 2021 data is not yet available, so selecting 2021 in the app will currently result in no data being displayed. There will also be null values for some questions when 2020 is selected, because some questions weren't asked in 2020. However, the application has been updated to handle null data gracefully, and this will be followed by #113, which will hide questions in the selector which are invalid for current year/location selections. 

Connects #112 

### Demo

<img width="1287" alt="Screen Shot 2021-10-25 at 4 41 12 PM" src="https://user-images.githubusercontent.com/21046714/138769860-4027a733-65a6-4c8a-87a0-d35d3c9ffc3a.png">

<img width="1278" alt="Screen Shot 2021-10-25 at 5 05 34 PM" src="https://user-images.githubusercontent.com/21046714/138770798-7b59d3b0-bee9-4e89-b65e-13288ee46e25.png">

<img width="1274" alt="Screen Shot 2021-10-25 at 4 39 43 PM" src="https://user-images.githubusercontent.com/21046714/138769877-b8e269b7-67d4-4e5c-8262-cd1e323c5ab5.png">

### Notes

We will still need to update to use the full 2021 dataset once it is available - see #119. However, proceeding with this task prior to the full dataset becoming available unblocks multiple other tasks, and the future data update should be quick (assuming no additional structural changes). 

## Testing Instructions

 * Download the two .xlsx files from [2021 data](https://drive.google.com/drive/folders/1yThtpb7ZwroeJXBfJCt3Y2b5Wi4DE0v5?usp=sharing), and place them in `/src/dataproc`
 * Follow the steps under `Data` and `Updating application with new data` in the Readme. The data processing script should succeed.
 * Run `./scripts/server` and confirm that the app runs as before without errors. 
 * Confirm that both years display in the year selection screen. 
 * Confirm that the new categories and questions are displayed on the questions page. 
 * Confirm that questions can be selected and their charts can be viewed as before (where data is available), and that null data doesn't cause the app to break.
